daml-lf/kv-support/src/main/scala/com/daml/lf/kv/archives/ArchiveConversions.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.kv.archives
+
+import com.daml.lf.archive.ArchiveParser
+import com.daml.lf.data.Ref
+import com.daml.lf.kv.ConversionError
+
+object ArchiveConversions {
+
+  def parsePackageId(rawArchive: RawArchive): Either[ConversionError.ParseError, Ref.PackageId] =
+    for {
+      archive <- ArchiveParser
+        .fromByteString(rawArchive.byteString)
+        .left
+        .map(error => ConversionError.ParseError(error.msg))
+      packageId <- Ref.PackageId
+        .fromString(archive.getHash)
+        .left
+        .map(ConversionError.ParseError)
+    } yield packageId
+
+  def parsePackageIdsAndRawArchives(
+      archives: List[com.daml.daml_lf_dev.DamlLf.Archive]
+  ): Either[ConversionError.ParseError, Map[Ref.PackageId, RawArchive]] =
+    archives.partitionMap { archive =>
```

Comment:
looks like a `traverse`

Reply:
We have an internal ban for cats etc. :)

Reply:
we don’t have a ban in `//daml-lf` though :)

Reply:
No need to ask twice, on my way :)

Reply:
This is what I hate about functional libraries. There is a warning about an unused import and a line that burns in red because of an allegedly wrong return type that I don't know how to fix. I can live with that. @cocreature is it better now?

Reply:
w8, I fixed `reconstructTransaction`, I will do the same here.

Reply:
Actually, this function is a lot shorter, so I would leave it like this, as I wouldn't like to use throw or scalaz (introduce false-positive warnings).